### ISSUE TITLE
Support risc-v

### ISSUE THREAD

**Issue by come-maiz** [ethereum/trin#1444]
RISC-V is a good start to get Ethereum nodes running in free hardware, which is a necessary step for a local-first decentralized economy. Lightclients are another necessary step, so these two worlds together are very powerful.

Please support the risc-v architecture in this project.

**Comment by come-maiz** [comment]
Next we have to update rustls, which is problematic because surf keeps it pinned to an old version and it is unmaintained:
https://github.com/http-rs/surf/issues/352#issuecomment-1709404352

**Comment by KolbyML** [comment]
> Next we have to update rustls, which is problematic because surf keeps it pinned to an old version and it is unmaintained: [http-rs/surf#352 (comment)](https://github.com/http-rs/surf/issues/352#issuecomment-1709404352)

Yeah I guess we should remove it as a dependency and switch to something else in that case

**Comment by KolbyML** [comment]
> Next we have to update rustls, which is problematic because surf keeps it pinned to an old version and it is unmaintained: [http-rs/surf#352 (comment)](https://github.com/http-rs/surf/issues/352#issuecomment-1709404352)

https://github.com/ethereum/trin/issues/1446

I moved a PR to move us to reqwest instead of surf, hopefull this helps with the mission

**Comment by come-maiz** [comment]
Nice! Thank you @KolbyML. It works.

Next is ethers-providers:
https://github.com/gakonst/ethers-rs/issues/2786

**Comment by KolbyML** [comment]
> Nice! Thank you @KolbyML. It works.
> 
> Next is ethers-providers: [gakonst/ethers-rs#2786](https://github.com/gakonst/ethers-rs/issues/2786)

We want to remove `etheres-rs` in Trin and replace it with `alloy`

**Comment by come-maiz** [comment]
That sounds perfect.

**Comment by KolbyML** [comment]
@come-maiz This PR https://github.com/ethereum/trin/pull/1456 removes our dependency on `ethers-rs`, are you able to compile on risc-v yet, or are there still some dependency issues?

**Comment by come-maiz** [comment]
thanks @KolbyML!

Compilation almost finishes now. It's getting stuck in librocksdb-sys, I'll see what I can do on my machine to get it to compile.

**Comment by KolbyML** [comment]
> thanks @KolbyML!
> 
> Compilation almost finishes now. It's getting stuck in librocksdb-sys, I'll see what I can do on my machine to get it to compile.

@come-maiz try installing clang and tell me if it works `sudo apt install clang`

**Comment by come-maiz** [comment]
It was installed.
